### PR TITLE
Add Newtonfool attack

### PR DESCRIPTION
**Pull request by zimmerrol**

### CODE REVIEW

**Comment by jonasrauber** [comment]
Can you please rename the attack to `NewtonFool` (with capital `F`) like in foolbox.
You can also remove the softmax implementation because I've added it to eagerpy (using the native implementations from TensorFlow / PyTorch).

**Comment by zimmerrol** [comment]
I have removed the softmax implementation and tidied up the NewtonFool implementation.

**Comment by jonasrauber** [comment]
Reminder to self: check if this implementation (of the gradient) is numerically stable

**Comment by jonasrauber** [comment]
merged the latest changes from master into this branch, to make the tests faster and coverage uploads successful
